diag_manager/fms_diag_field_object.F90

```diff
@@ -359,7 +368,115 @@ subroutine set_vartype(objin , var)
           " r8, r4, i8, i4, or string.", warning)
  end select
 end subroutine set_vartype
-
+!> Allocates the data buffer in the field object.
+!! Adds the input data to the buffered data.
+subroutine set_data_buffer (this, input_data, diag_axis)
+  class (fmsDiagField_type) , intent(inout):: this !< The field object
+  class(*), dimension(:,:,:,:), intent(in) :: input_data !< The input array
+  class(fmsDiagAxisContainer_type),intent(in)   :: diag_axis(:)          !< Array of diag_axis
+!> Allocate the buffer if it is not allocated
+  if (.not.allocated(this%data_buffer_allocated)) this%data_buffer_allocated = .false.
+  if (.not.this%data_buffer_allocated) &
+    this%data_buffer_allocated =  allocate_data_buffer(this, input_data, diag_axis)
+  if (.not.this%data_buffer_allocated) &
+    call mpp_error ("set_data_buffer", "The data buffer for the field "//trim(this%varname)//" was unable to be "//&
+      "allocated.", FATAL)
+
+!> Buffer a copy of the data
+  select type (input_data)
+    type is (real(kind=r4_kind))
+      select type (db => this%data_buffer)
+        type is (real(kind=r4_kind))
+          db = input_data
```

Comment:
`input_data` contains the data for the 1 thread (only 1 section of the global data)
this%data_buffer is allocated to be the size of the global data

so this is not going to work because the sizes may not be the same for more than 1 thread.

so we need to pass in `is_in, js_in, ks_in, ie_in, je_in, ke_in` from fms_diag_accept_data then do
```F90
db(is_in:ie_in, js_in:je_in, ks_in:ke_in, :) = input_data
```

Reply:
I updated it to look like how you have it.  Do I need to put the indicies on input_data?  I don't think so.

Reply:
I don’t think so